### DISCUSSION TITLE
How to enable SPICE Guest tools on macOS Guest?

### DISCUSSION THREAD

**Discussion by ghost** [utmapp/UTM#6833] [UTM Help (macOS)] [answered]
> (macOS 15) Improved macOS guest support The last window size (including full screen size) will be restored when the guest supports dynamic resolution. Removable drives and shared directories can now be ejected and changed while the macOS VM is running. Copy/paste synchronization between macOS 15 guest and host now works when the guest tools are installed (from the CD icon in the toolbar).

I don't see an option to install it in the guest. I am on UTM 4.6.1 and using an IPSW file that I sourced from outside UTM.

**Comment by ghost** [comment]
Alright, so I manually installed [utm-guest-tools-macos-0.22.1.img](https://github.com/utmapp/vd_agent/releases/download/spice-vdagent-0.22.1-macOS/utm-guest-tools-macos-0.22.1.img) and added it a removable drive to my existing macOS guest VM. I then opened the `GUEST TOOLS` image and installed `spice-vdagent-0.22.1.pkg` but clipboard sharing between the macOS guest and the macOS host still did not work. Both the guests and the host are on macOS 15.1.1

**Comment by ghost** [comment]
Does the `spice-vdagent-0.22.1.pkg` need to be installed on the host too?

**Comment by osy** [comment]
If you are on macOS 15, it should be in the removable drives menu (CD icon). Once it's installed in the guest, you should get two prompts to allow spice-vdagentd and spice-vdagent to launch. Once you confirm it should work.

**Reply by ghost** [reply]
Okay so I got it to work but two things to note:

- You need to run `spice-vdagent-0.22.1.pkg` and complete the installation in an administrator user within the guest otherwise it simply will not work. You can then switch to a standard user inside the guest and clipboard sharing will work fine. Could be something to add in the documentation.

- Secondly, I was still unable to get the package from the `removable drives menu`. There is no CD icon. Could you send me a screenshot of what I should be seeing? This is what I see on a brand new macOS 15 install: 
![screenshot](https://github.com/user-attachments/assets/b869b41d-d731-49ee-ab9a-539dc02cf569)


I have searched incredibly thoroughly in the UTM menu and tried many different things but the SPICE tools do not install by themselves at all and are not present inside the guest. Could this require an administrator host user? I am currently using a standard host user.

 I have also installed UTM to the local application directory rather than the system one, would this make a difference?

**Comment by osy** [comment, marked as answer]
@duck09 
<img width="395" alt="image" src="https://github.com/user-attachments/assets/3dde4cf6-0ae6-4a92-ac6a-36d04ad82bf7">

Do you not see this?

**Reply by ghost** [reply]
Ah I found them. I never noticed these settings before given that they only appear once the VM is running. They are kindof hard to find. Thanks anyway!

**Reply by benzfy** [reply]
<img width="247" alt="image" src="https://github.com/user-attachments/assets/ca662005-4b55-4d92-9ee7-f55dbaac56dc" />

OMG, this it the "CD icon". I thought it was a pokeball or something.

Thank you very much.

**Reply by mbayo-ise** [reply]
this option is not selectable for me
![image](https://github.com/user-attachments/assets/42e5f0a3-43f5-4c08-a01d-5de5338021b8)

**Reply by rublev** [reply]
> this option is not selectable for me ![image](https://private-user-images.githubusercontent.com/98422243/455616860-42e5f0a3-43f5-4c08-a01d-5de5338021b8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg1NTg5MzAsIm5iZiI6MTc2ODU1ODYzMCwicGF0aCI6Ii85ODQyMjI0My80NTU2MTY4NjAtNDJlNWYwYTMtNDNmNS00YzA4LWEwMWQtNWRlNTMzODAyMWI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTE2VDEwMTcxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxYTQzNDUwYzNjZGI5OGExNDM3ZGE1MjViODk0MzJmODk5ZWI0ZDc0YmNhYjhkMWRmZTZiNjAwZTU3NTM3ZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-b3veTUMyxfGLmCjxpiV__Jy4pG5Z7mo-2DAIWzNTfA)

same!! 😭

**Reply by levietanh1809** [reply]
OMG. Thank u somuch

**Comment by ArturLinhart** [comment]
For anyone looking for an Intel x86_64 build of spice-vdagent 0.22.1 for an older macOS guest:

I built and tested an unofficial x86_64 version targeting macOS 10.14 Mojave and later.

Tested with:

macOS Mojave 10.14.6 x86_64 guest
QEMU/KVM 7.2.22
SPICE virtio-serial channel com.redhat.spice.0
clipboard sharing in both directions

The official upstream .pkg is still used first to install the launchd configuration; only the spice-vdagent and spice-vdagentd binaries are replaced.

Build, binaries, installation instructions and tested libvirt configuration:

https://github.com/Algites-EU/pub.tool.CustomBuilds/releases/tag/spice-vdagent-0.22.1-macos-x86_64-mojave-1

This is an unofficial community build and is not affiliated with or supported by the UTM/vd_agent project.
